sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -660,10 +688,63 @@ trait ScalaReflection {
       case t if t <:< definitions.ByteTpe => Schema(ByteType, nullable = false)
       case t if t <:< definitions.BooleanTpe => Schema(BooleanType, nullable = false)
       case other =>
-        throw new UnsupportedOperationException(s"Schema for type $other is not supported")
+        val fields = getJavaBeanFields(other)
+        if (fields.length > 0) {
+          Schema(StructType(
+            fields.map { f =>
+              val Schema(dataType, nullable) = schemaFor(classToType(f.getType))
+              StructField(f.getName, dataType, nullable)
+            }), nullable = true)
+        } else {
+          throw new UnsupportedOperationException(s"Schema for type $other is not supported")
+        }
     }
   }
 
+  private def classToType(cls: Class[_]): Type = cls match {
+    case c if c == classOf[Boolean] => localTypeOf[Boolean]
+    case c if c == classOf[Byte] => localTypeOf[Byte]
+    case c if c == classOf[Short] => localTypeOf[Short]
+    case c if c == classOf[Int] => localTypeOf[Int]
+    case c if c == classOf[Long] => localTypeOf[Long]
+    case c if c == classOf[Float] => localTypeOf[Float]
+    case c if c == classOf[Double] => localTypeOf[Double]
+    case c if c.isArray =>
+      universe.appliedType(localTypeOf[Array[_]], classToType(c.getComponentType) :: Nil).erasure
```

Comment:
I came up with this by intuition and verified it by experiment, cannot find any doc for it, very hacky...